src/main/scala/edu/ie3/osmogrid/io/input/InputDataProvider.scala

```diff
@@ -6,31 +6,136 @@
 
 package edu.ie3.osmogrid.io.input
 
-import akka.actor.typed.Behavior
-import akka.actor.typed.ActorRef
-import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors, StashBuffer}
+import akka.actor.typed.{ActorRef, Behavior}
+import com.acervera.osm4scala.EntityIterator.fromPbf
+import com.acervera.osm4scala.model.{NodeEntity, RelationEntity, WayEntity}
 import edu.ie3.osmogrid.cfg.OsmoGridConfig
 import edu.ie3.osmogrid.guardian.OsmoGridGuardian.OsmoGridGuardianEvent
+import edu.ie3.osmogrid.model.{OsmoGridModel, PbfFilter}
+import edu.ie3.util.osm.model.OsmEntity.{Node, Relation, Way}
+import org.locationtech.jts.geom.{
+  Coordinate,
+  LinearRing,
+  Point,
+  Polygon,
+  PrecisionModel
+}
+
+import java.io.{FileInputStream, InputStream}
+import java.time.ZonedDateTime
+import java.util.UUID
+import scala.collection.mutable.ListBuffer
+import scala.util.{Failure, Success, Try, Using}
 
 object InputDataProvider {
 
+  // external requests
+  sealed trait InputRequest
+
+  final case class ReqOsm(
+      runId: UUID,
+      replyTo: ActorRef[InputDataProvider.Response],
+      filter: PbfFilter
+  ) extends InputRequest
+      with InputDataEvent
+
+  final case class ReqAssetTypes(
+      runId: UUID,
+      replyTo: ActorRef[InputDataProvider.Response]
+  ) extends InputRequest
+      with InputDataEvent
+
+  case object Terminate extends InputRequest with InputDataEvent
+
+  // external responses
+  sealed trait Response
+  final case class RepOsm(runId: UUID, osmModel: OsmoGridModel) extends Response
+  final case class InvalidOsmRequest(reqRunId: UUID, actualRunId: UUID)
+      extends Response
+  final case class OsmReadFailed(reason: Throwable)
+      extends Response
+      with InputDataEvent
+  final case class RepAssetTypes(osmModel: OsmoGridModel) extends Response
```

Comment:
Are you sure about the returned data type? :wink:

Reply:
Not quite sure what you mean here, could you clarify?